mantis/acquisition/acq_engine.py

```diff
@@ -278,64 +301,109 @@ def setup(self, output_path: Union[str, os.PathLike] = None):
+                        shard_size_chunks=self.zarr_settings.shard_sizes['z'],
+                        kind=DimensionType.SPACE,
+                    ),
+                    Dimension(
+                        name='y',
+                        array_size_px=y_size,
+                        chunk_size_px=min(self.zarr_settings.chunk_sizes['y'], max(1, y_size)),
+                        shard_size_chunks=self.zarr_settings.shard_sizes['y'],
+                        kind=DimensionType.SPACE,
+                    ),
+                    Dimension(
+                        name='x',
+                        array_size_px=x_size,
+                        chunk_size_px=min(self.zarr_settings.chunk_sizes['x'], max(1, x_size)),
+                        shard_size_chunks=self.zarr_settings.shard_sizes['x'],
+                        kind=DimensionType.SPACE,
+                    ),
+                ]
 
                 if self.channel_settings.num_channels > 1:
-                    zarr_settings.dimensions.insert(
+                    dimensions.insert(
                         1,
-                        aqz.Dimension(
+                        Dimension(
                             name='c',
                             array_size_px=self.channel_settings.num_channels,
-                            chunk_size_px=int(self.channel_settings.num_channels),
-                            shard_size_chunks=1,
-                            kind=aqz.DimensionType.CHANNEL,
+                            chunk_size_px=min(
+                                self.zarr_settings.chunk_sizes['c'],
+                                max(1, self.channel_settings.num_channels),
+                            ),
+                            shard_size_chunks=self.zarr_settings.shard_sizes['c'],
+                            kind=DimensionType.CHANNEL,
                         ),
                     )
-                self._zarr_writer = aqz.ZarrStream(zarr_settings)
+
+                # Create array settings using ZarrSettings
+                array_settings = ArraySettings(
+                    dimensions=dimensions,
+                    data_type=self.zarr_settings.get_data_type_enum(),
+                )
+
+                # Set store path in zarr_settings if not already set
+                if self.zarr_settings.store_path is None:
+                    self.zarr_settings.store_path = str(output_path)
+
+                # Create stream settings with the array
+                if self.zarr_settings.use_hcs_layout:
+                    # Create HCS layout with plate and wells
+                    plate = Plate(
+                        name=self.zarr_settings.plate_name,
+                        description=self.zarr_settings.plate_description or "",
+                    )
+
+                    # Create wells from position settings
+                    wells = []
+                    for well_id in set(self.position_settings.well_ids):
+                        well = Well(
```

Comment:
Let's make the default well `0/0` rather than `A/1`. We can work on improving this parsing in the future, seem fragile for now